extensions-core/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleStorage.java

```diff
@@ -159,11 +167,21 @@ public void delete(final String bucket, final String path) throws IOException
    * @param bucket GCS bucket
    * @param paths  Iterable for absolute paths of objects to be deleted inside the bucket
    */
-  public void batchDelete(final String bucket, final Iterable<String> paths) throws IOException
+  public void batchDelete(final String bucket, final Iterable<String> paths)
   {
-    List<Boolean> statuses = storage.get().delete(Iterables.transform(paths, input -> BlobId.of(bucket, input)));
+    final List<Boolean> statuses = storage.get().delete(Iterables.transform(paths, input -> BlobId.of(bucket, input)));
+    final List<String> failedPaths = new ArrayList<>();
+    final Iterator<String> pathIterator = paths.iterator();
+    int cursor = 0;
+    while (pathIterator.hasNext()) {
+      String failedPath = pathIterator.next();
+      if (!statuses.get(cursor)) {
+        failedPaths.add(failedPath);
+      }
+      cursor++;
+    }
```

Comment:
All of it should be moved inside 
`if log.isDebug()`

Reply:
Thanks. Oversight on my part.